indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -2291,9 +2291,30 @@ protected boolean supportsPartitionExpiration()
     return false;
   }
 
+  protected boolean shouldSkipIgnorablePartitions()
```

Comment:
Should we not do something similar for Kafka? Why is this not an issue with Kafka?

Reply:
We haven't encountered something similar in Kafka yet. But the API has been put in place so that if `KafkaSupervisor` needs to do something similar, it can override the methods `shouldSkipIgnorablePartitions()` and `getIgnorablePartitionIds()`.

For Kinesis, the ignorable partitions translate to empty and closed shards, which is a concept specific to Kinesis.